File: BootCampOOP/Basic/Vector/Vector.cpp
#include "Vector.h"

Vector::Vector(double xIn, double yIn) {
    this->x = xIn;
    this->y = yIn;
}

Vector::~Vector() {}

double Vector::getX() const {
    return x;
}

double Vector::getY() const {
    return y;
}

void Vector::setX(double value) {
    x = value;   
}

void Vector::setY(double value) {
    y = value;
}

double Vector::len() const{
    return hypot(x,y);
}

bool Vector::operator==(const Vector& other) const {
    return (x == other.getX() && y == other.getY());
}

bool Vector::operator!=(const Vector& other) const {
    return !operator==(other);
}

void Vector::operator+=(const Vector& other) {
    x += other.getX();
    y += other.getY();
}

void Vector::operator-=(const Vector& other) {
    x -= other.getX();
    y -= other.getY();
}

Vector Vector::operator+(const Vector& other) const {
    Vector result;
    result.x = this->x + other.getX();
    result.y = this->y + other.getY();
    
    return result;
}

Vector Vector::operator-(const Vector& other) const {
    Vector result;
    result.x = this->x - other.getX();
    result.y = this->y - other.getY();
    
    return result;
}

std::ostream& operator<<(std::ostream& out, const Vector& vector) {
    out << "(" << vector.getX() << ", " << vector.getY() << ")";
    return out;
}

File: BootCamp Java/Basic/Vector/Vector.java
public class Vector {
    private double x;
    private double y;
    
    public Vector( double x, double y) {
        this.x = x;
        this.y = y;
    }
    public  Vector() {
        this( 0, 0);
    }
    public double getX() {
        return this.x;
    }

    public double getY() {
        return this.y;
    }

    public void setX(double value) {
        this.x = value;
    }

    public  void setY(double value) {
    this.y = value;
    }
    public double len() {
        return Math.hypot(this.x,this.y);
    }

    public boolean isEqual( Vector other) {
        return this.x == other.x && this.y == other.y;
    }
    public boolean isNotEqual(Vector other) {
        return !isEqual(other);
    }

    public void increment(Vector other) {
        this.x += other.x; 
        this.y += other.y;
    }

    public void decrement(Vector other) {
        this.x -= other.x; 
        this.y -= other.y;
    }

    public Vector sum(Vector other) {
        Vector sum = new Vector();
        
        sum.x = this.x + other.x;
        sum.y = this.y + other.y;
        return sum;
    }

    public Vector dif(Vector other) {
            Vector dif = new Vector();
        
        dif.x = this.x - other.x;
        dif.y = this.y - other.y;
        return dif;
    }
}